### PR TITLE
executor: fix cases that some intermediate errors may be ignored when converting a vector to bools

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Fix the cases that when evaluating an expression and converting its values to bools, some intermediate errors may be ignored.

### What is changed and how it works?
Return the error immediately when it occurs.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/bench_test.go [hunk 1]
```diff
@@ -1592,6 +1592,15 @@ func (s *testEvaluatorSuite) TestVecEvalBool(c *C) {
 	}
 }
 
+func (s *testEvaluatorSuite) TestVecToBool(c *C) {
+	ctx := mock.NewContext()
+	buf := chunk.NewColumn(eType2FieldType(types.ETString), 2)
+	buf.ReserveString(2)
+	buf.AppendString("999999999999999999923")
```
Comment (SunRunAway) [inline expression/bench_test.go]: We need assert that `types.StrToInt("999999999999999999923")` will meet error.

Comment (qw4990) [reply expression/bench_test.go]: This statement `c.Assert(toBool(ctx.GetSessionVars().StmtCtx, types.ETString, buf, []int{0, 1}, []int8{0, 0}), NotNil)` guarantees there is an error.

Comment (qw4990) [reply expression/bench_test.go]: Updated, PTAL.

File: expression/expression.go [hunk 2]
```diff
@@ -186,6 +186,17 @@ func IsEQCondFromIn(expr Expression) bool {
 	return len(cols) > 0
 }
 
+// HandleOverflowOnSelection handles Overflow errors when evaluating selection filters.
+// We should ignore overflow errors when evaluating selection conditions:
+//		INSERT INTO t VALUES ("999999999999999999");
+//		SELECT * FROM t WHERE v;
+func HandleOverflowOnSelection(sc *stmtctx.StatementContext, val int64, err error) (int64, error) {
+	if sc.InSelectStmt && err != nil && types.ErrOverflow.Equal(err) {
+		return -1, nil
```
Comment (zz-jason) [inline expression/expression.go]: could you add a test case for this code branch?

Comment (qw4990) [reply expression/expression.go]: `TestIssue14159` will go to this branch.

**Pull request merged**